drivers/sensor/iis2iclx/iis2iclx_trigger.c

```diff
@@ -228,18 +221,18 @@ int iis2iclx_init_interrupt(const struct device *dev)
 	iis2iclx->work.handler = iis2iclx_work_cb;
 #endif /* CONFIG_IIS2ICLX_TRIGGER_OWN_THREAD */
 
-	ret = gpio_pin_configure(iis2iclx->gpio, cfg->irq_pin,
-				 GPIO_INPUT | cfg->irq_flags);
+	ret = gpio_pin_configure(cfg->gpio_drdy.port, cfg->gpio_drdy.pin,
+				 GPIO_INPUT | cfg->gpio_drdy.dt_flags);
 	if (ret < 0) {
 		LOG_ERR("Could not configure gpio");
 		return ret;
 	}
 
 	gpio_init_callback(&iis2iclx->gpio_cb,
 			   iis2iclx_gpio_callback,
-			   BIT(cfg->irq_pin));
+			   BIT(cfg->gpio_drdy.pin));
 
-	if (gpio_add_callback(iis2iclx->gpio, &iis2iclx->gpio_cb) < 0) {
+	if (gpio_add_callback(cfg->gpio_drdy.port, &iis2iclx->gpio_cb) < 0) {
```

Comment:
I don't remember what color we chose for this bikeshed; does anything in the driver infrastructure verify that the passed driver instance is ready?  @galak @mnkp

I see it being checked in the clock control driver glue code, but not gpio.

Reply:
I'm checking 

```if(device_is_ready(cfg->gpio_drdy.port))```

at the beginning of the interrupt init function. Is this your concern? I'm not sure I got the point correctly.

Reply:
OK, it's fine here, but in general requiring this sort of check at all call sites is error-prone.  But maybe that's what people wanted.  I'm not blocking on this.

Reply:
> does anything in the driver infrastructure verify that the passed driver instance is ready?

I don't think so. We check that the dev pointer passed to the function call is not NULL if user mode is enabled.